cluster-autoscaler/cloudprovider/azure/azure_cloud_provider_test.go

```diff
@@ -135,6 +135,7 @@ func TestNodeGroupForNode(t *testing.T) {
 
 	registered := provider.azureManager.RegisterNodeGroup(
 		newTestScaleSet(provider.azureManager, "test-asg"))
+	provider.azureManager.explicitlyConfigured["test-asg"] = true
```

Comment:
this is needed because the manager setup logic skips `createAzureManagerInternal` and thus by the time `fetchAutoNodegroups` is called `explicitlyConfigured` will be empty and the nodegroup will be unregistered on L217 in azure_manager.